### ISSUE TITLE
Error Status 400

### ISSUE THREAD

**Issue by apexJCL** [realm/realm-mobile-platform#202]
[See StackOverflow's question](https://stackoverflow.com/questions/44170810/realm-response-status-400)

I'm trying to share a realm between users for a proof-of-concept messaging app on Android.

I have a realm with access url: `realm://192.168.1.90:9080/lynxenger/contacts`, the realm exists and has one register, but when I try to change it's permissions using an admin user that created the realm, I get the following on the log:

` E/REALM_SYNC: Connection[2]: HTTP: Failed to switch protocols (response status = 400)`

[Here's a Gist with the code][1] I use to change permissions and update some data 

  [1]: https://gist.github.com/apexJCL/db57d140c573d311131cc33050113c3f

There seems to be an error happening at server-side, here's the log file when the error occurs:

```sh
[proxy] internal error. Code undefined, status: undefined, message: SyntaxError: Unexpected token j in JSON at position 0
    at Object.parse (native)
    at ProxyService.webSocketUpgradeHandler (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:104:30)
    at ProxyService.safeUpgrade (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:78:12)
    at emitThree (events.js:116:13)
    at Server.emit (events.js:194:7)
    at onParserExecuteCommon (_http_server.js:411:14)
    at HTTPParser.onParserExecute (_http_server.js:379:5)
```
[Here's a pastebin](https://pastebin.com/3yjvxJNV) with the android log, **I removed** the user's token, to avoid long lines, but if needed I can put it on here

**Comment by mokkun** [comment]
I'm having a similar issue (different steps, same outcome). Whenever I try to create the realm using an admin user, I get the mentioned internal error. Using a normal user works just fine.
```
2017-05-27T17:59:08.396Z - warn: [proxy] internal error. Code undefined, status: undefined, message: SyntaxError: Unexpected token j in JSON at position 0
    at Object.parse (native)
    at ProxyService.webSocketUpgradeHandler (/Users/mokkun/Developer/realm-mobile-platform/realm-object-server/node_modules/realm-object-server-developer/.build/src/node/services/proxy.js:104:30)
    at ProxyService.safeUpgrade (/Users/mokkun/Developer/realm-mobile-platform/realm-object-server/node_modules/realm-object-server-developer/.build/src/node/services/proxy.js:78:12)
    at emitThree (events.js:116:13)
    at Server.emit (events.js:194:7)
    at onParserExecuteCommon (_http_server.js:411:14)
    at HTTPParser.onParserExecute (_http_server.js:379:5)
```

**Comment by bigfish24** [comment]
Thanks for filling this, we have an engineer focusing on figuring this out.

**Comment by mrackwitz** [comment]
Further investigations confirm that this is a bug in the Java client SDK. We're working on it. Please refer to https://github.com/realm/realm-java/issues/4750 for more info.

**Comment by cmelchior** [comment]
This has been fixed with Realm Java 3.3.2
